### PR TITLE
Dhcp keywords 5506 v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5506
https://redmine.openinfosecfoundation.org/issues/5507

Describe changes:
- Adds a DHCP rebinding_time keyword
- Adds a DHCP renewal_time keyword

suricata-verify-pr: 919
https://github.com/OISF/suricata-verify/pull/919

### CODE REVIEW

**Comment by victorjulien** [comment]
Do these support prefilter?

**Comment by catenacyber** [comment]
> Do these support prefilter?

No, can it be done for app-layer keywords ?

The prefilter API takes `Packet` as input, but no tx... cf `PrefilterPacketIcmpv6mtuMatch` for instance

**Comment by victorjulien** [comment]
> > Do these support prefilter?
> 
> No, can it be done for app-layer keywords ?
> 
> The prefilter API takes `Packet` as input, but no tx... cf `PrefilterPacketIcmpv6mtuMatch` for instance

There is a TX version of the API, see `PrefilterAppendTxEngine` etc. But if it is fully complete from this PoV, I'm not sure.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on ips_afp_drop_chk.
 | field | test | baseline | % |
|--|--|--|--| 
| | ips_afp_stats_chk | | |
| .flow.end.state.new | 13463 | 10800 | 124.66% |
| .flow.end.tcp_liberal | 124049 | 108000 | 114.86% |
| .tcp.reassembly_gap | 157004 | 108000 | 145.37% |
| | generic_stats_chk | | |
| .capture.kernel_drops | 4735537 | 5654519 | 83.75% |
| .flow.end.tcp_state.syn_sent | 642 | 183 | 350.82% |
| .flow.end.tcp_state.syn_recv | 34 | 52 | 65.38% |
| .tcp.segment_memcap_drop | 27275 | 11729 | 232.54% |
| .tcp.reassembly_gap | 135831 | 114099 | 119.05% |
| .tcp.insert_data_normal_fail | 27008 | 11358 | 237.79% |
| .app_layer.error.http.parser | 76 | 55 | 138.18% |
| .app_layer.error.smtp.gap | 48 | 61 | 78.69% | 

Pipeline 8799
WARNING: THERE IS A KNOWN BAD BASELINE WITH PACKET DROPS. bE MINDFUL OF ANY RESULTS.

File: doc/userguide/rules/dhcp-keywords.rst [hunk 1]
```diff
@@ -19,3 +19,41 @@ The version can be matched exactly, or compared using the _op_ setting::
...
+dhcp.rebinding_time
+-------------------
+
+DHCP rebinding time (integer).
+
+Syntax::
+
+ dhcp.rebinding_time:[op]<number>
+
+The version can be matched exactly, or compared using the _op_ setting::
```
Comment (jufajardini) [inline doc/userguide/rules/dhcp-keywords.rst:32]: I believe that here and in the other places where `version` is mentioned here, we are actually talking about the specific keywords described, right?

**Review by jufajardini — CHANGES_REQUESTED** [review]
Doc needs just one change, from me :)

**Comment by catenacyber** [comment]
> There is a TX version of the API, see `PrefilterAppendTxEngine` etc. But if it is fully complete from this PoV, I'm not sure.

So I have been looking at this.
And I have been looking at `DetectRunPrefilterTx` which seems to work only for pattern matching...
cf `engine->cb.PrefilterTx` which has no return value and the magic is done with `mpm_table[mpm_ctx->mpm_type].Search` with `det_ctx->pmq` stuff...

What would be the idea of pre filtering for integer keywords about transactions ?
Like should it run all signatures using this keyword in one go ?
Or can we somehow simply run `AppLayerTxMatch = DetectDHCPLeaseTimeMatch` ?

File: doc/userguide/rules/dhcp-keywords.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/dhcp-keywords.rst:32]: Thanks :-)

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7804

**Pull request closed without merging**